### PR TITLE
Allow to use selectors with volumeClaimTemplates

### PR DESCRIPTION
**Pull request by claudio-vellage**
### Motivation

Currently it's not possible to use selectors with volumeClaimTemplates which makes it hard/impossible to bind statically provisioned PVs.

### Modifications

Added (optional) selectors to `volumeClaimTemplates` and documented in values file.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### CODE REVIEW

**Comment by claudio-vellage** [comment]
@lhotari is there anything left to do for me? Can this be assigned to someone for review? It's a rather small change which doesn't affect existing deployments, only adds additional functionality. It allows that selectors can be added to volumeClaimTemplates, which will allow to easily create PVCs which bind to existing PVs using selectors. This allows to reuse the same PVs even if all pulsar resources are deleted. This will be more robust for production deployments and avoid having to restore the content of the PVs manually in case of disaster recovery.

File: charts/pulsar/values.yaml [hunk 1]
```diff
@@ -383,6 +383,10 @@ zookeeper:
         # type: pd-ssd
         # fsType: xfs
         # provisioner: kubernetes.io/gce-pd
+      ## If you want to bind static persistent volumes via selectors, e.g.:
+      # selectors:
+        # matchLabels:
+        # app: pulsar-zookeeper
```
Comment (michaeljmarshall) [inline charts/pulsar/values.yaml:389]: I think it is slightly more helpful to have a good zero value here. Also, we need `selector` not `selectors`.

```suggestion
      ## If you want to bind static persistent volumes via selectors, e.g.:
      selector: {}
```

File: charts/pulsar/values.yaml [hunk 2]
```diff
@@ -515,6 +519,10 @@ bookkeeper:
         # type: pd-ssd
         # fsType: xfs
         # provisioner: kubernetes.io/gce-pd
+      ## If you want to bind static persistent volumes via selectors, e.g.:
+      # selectors:
+        # matchLabels:
+        # app: pulsar-bookkeeper-journal
```
Comment (michaeljmarshall) [inline charts/pulsar/values.yaml:525]:
```suggestion
      selector: {}
```

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
@claudio-vellage - thank you for your contribution. Sorry for the delayed review here. Once we get the `values.yaml` cleaned up, I plan to merge this. Thanks.

File: charts/pulsar/values.yaml (hunk 2 above)
Comment (claudio-vellage) [reply charts/pulsar/values.yaml]: I've made the changes, please review

File: charts/pulsar/values.yaml (hunk 1 above)
Comment (claudio-vellage) [reply charts/pulsar/values.yaml]: I've made the changes, please review

**Review by michaeljmarshall — APPROVED** [review]
LGTM

**Comment by michaeljmarshall** [comment]
@claudio-vellage - are you able to resolve the merge conflicts? Thanks!

**Comment by michaeljmarshall** [comment]
Closing and reopening to get the latest test changes. (I am pretty sure we don't need to merge master into this branch to get the fix I just merge here #315, but that might be necessary.)

**Pull request merged**